addons/pos_adyen/controllers/main.py

```diff
@@ -19,14 +19,20 @@ def notification(self):
 
         _logger.info('notification received from adyen:\n%s', pprint.pformat(data))
         terminal_identifier = data['SaleToPOIResponse']['MessageHeader']['POIID']
-        payment_method = request.env['pos.payment.method'].sudo().search([('adyen_terminal_identifier', '=', terminal_identifier)], limit=1)
+        payment_method_sudo = request.env['pos.payment.method'].sudo().search([('adyen_terminal_identifier', '=', terminal_identifier)], limit=1)
 
-        if payment_method:
+        pos_order_id = data["SaleToPOIResponse"]["PaymentResponse"]["SaleData"]["SaleTransactionID"]["TransactionID"]
+        pos_session_id = int(pos_order_id[:5])
+        pos_config_sudo = request.env["pos.session"].sudo().browse(pos_session_id).config_id
+
+        if payment_method_sudo:
             # These are only used to see if the terminal is reachable,
             # store the most recent ID we received.
             if data['SaleToPOIResponse'].get('DiagnosisResponse'):
-                payment_method.adyen_latest_diagnosis = data['SaleToPOIResponse']['MessageHeader']['ServiceID']
+                payment_method_sudo.adyen_latest_diagnosis = data['SaleToPOIResponse']['MessageHeader']['ServiceID']
             else:
-                payment_method.adyen_latest_response = json.dumps(data)
+                stringified_data = json.dumps(data)
+                payment_method_sudo.adyen_latest_response = stringified_data
+                request.env["bus.bus"].sudo()._sendone(f"pos_config-{pos_config_sudo.id}", "ADYEN_LATEST_RESPONSE", stringified_data)
```

Comment:
Instead of sending the full information, you should just notify the UI that the payment has been validated. Then, from the UI, make an extra rpc to get the full payment information. This is reduce security risk when using the websocket.

Reply:
@caburj can you take a look at the new commit?